src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -371,6 +358,23 @@ private string GetRelativePath(RouteTemplate parsedTemplate)
         {
             var results = new List<ApiResponseFormat>();
 
+            // Build list of all possible return types (and status codes) for an action.
+            var responseTypes = new Dictionary<int, Type>();
+
+            // If the action returns a type, then automatically consider it as a possible api response format with
+            // a status code 200 OK.
+            // NOTE: In some cases this might sound incorrect, for example in the case where the action is returning
+            // 'void' or 'Task'. We want to keep it this way to enable users to override the type via the response
+            // metadata attributes. So an exmple is that action returns void but there is a Produces attribute which
+            // sets the response type to be 'Produces'
```

Comment:
No, I don't think we should allow this case with `void` and `Task`. If that's the case, let them set the status code, but have the format be `void`